workflow/controller/artifact_gc.go

```diff
@@ -154,7 +154,7 @@ func (woc *wfOperationCtx) processArtifactGCStrategy(ctx context.Context, strate
 		}
 		template, found := templatesByName[templateName]
 		if !found {
-			template = woc.execWf.GetTemplateByName(templateName)
```

Comment:
This function uses Workflow "Status", which is not necessarily made current in `execWf`, so we need to use `wf`. It's interesting that previous tests involving Workflows based on WorkflowTemplates have passed. That must be because the Status is incorporated into the `execWf` object each time the wfOperationCtx is [constructed](https://github.com/argoproj/argo-workflows/blob/master/workflow/controller/operator.go#L151).